src/angular/hq/src/app/projects/project-create/project-create.component.ts

```diff
@@ -1,12 +1,197 @@
+import { PdfViewerModule } from 'ng2-pdf-viewer';
 import { CommonModule } from '@angular/common';
 import { Component } from '@angular/core';
+import {
+  BehaviorSubject,
+  Observable,
+  debounceTime,
+  firstValueFrom,
+  map,
+  shareReplay,
+  switchMap,
+} from 'rxjs';
+import { GetStaffV1Record } from '../../models/staff-members/get-staff-member-v1';
+import { HQService } from '../../services/hq.service';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+  ValidationErrors,
+  Validators,
+} from '@angular/forms';
+import { ClientListComponent } from '../../clients/client-list/client-list.component';
+import { GetQuotesRecordV1 } from '../../models/quotes/get-quotes-v1';
+import { APIError } from '../../errors/apierror';
+import { GetClientRecordV1 } from '../../models/clients/get-client-v1';
+import { SelectableClientListComponent } from '../../clients/selectable-client-list/selectable-client-list.component';
+import { UpsertProjectRequestV1 } from '../../models/projects/upsert-project-v1';
+import { Router, ActivatedRoute } from '@angular/router';
+
+export enum Period {
+  Week = 1,
+  Month = 2,
+  Quarter = 3,
+  Year = 4,
+}
 
 @Component({
   selector: 'hq-project-create',
   standalone: true,
-  imports: [CommonModule],
-  templateUrl: './project-create.component.html'
+  imports: [
+    CommonModule,
+    FormsModule,
+    ReactiveFormsModule,
+    SelectableClientListComponent,
+    PdfViewerModule,
+  ],
+  templateUrl: './project-create.component.html',
 })
 export class ProjectCreateComponent {
-  modalOpen = false;
+  projectManagers$: Observable<GetStaffV1Record[]>;
+  quotes$: Observable<GetQuotesRecordV1[]>;
+  selectedQuote$ = new Observable<string>();
+  quotePdfURL = 'https://vadimdez.github.io/ng2-pdf-viewer/assets/pdf-test.pdf';
+
+  apiErrors: string[] = [];
+  selectedClientName$ = new BehaviorSubject<string | null>(null);
+  generatedChargeCode$ = new BehaviorSubject<string | null>(null);
+
+  projectFormGroup = new FormGroup(
+    {
+      clientId: new FormControl(''),
+      name: new FormControl('', Validators.required),
+      projectManagerId: new FormControl(''),
+      hourlyRate: new FormControl(0, [Validators.required, Validators.min(0)]),
+      totalHours: new FormControl('', [Validators.required, Validators.min(0)]),
+      bookingPeriod: new FormControl(0, [
+        Validators.required,
+        Validators.min(0),
+      ]),
+      quoteId: new FormControl('', Validators.required),
+      startDate: new FormControl(new Date(), Validators.required),
+      endDate: new FormControl(new Date(), Validators.required),
+    },
+    { validators: this.dateRangeValidator }
+  );
+
+  // PDF VIEWER
+  page: number = 1;
+  totalPages?: number;
+  isLoaded: boolean = false;
```

Comment:
Instead of dropping the PDF viewer logic inside the project create component, I want to create a dedicated HQ PDF component that wraps the library. That way, if we need to swap the library in the future we can without having to replace it in multiple places. I expect we will probably be using the PDF functionality on the quote details page.